pkg/kudoctl/cmd/install/install.go

```diff
@@ -46,33 +57,44 @@ func verifyFrameworks(args []string) error {
 		return fmt.Errorf("--package-version not supported in multi framework install")
 	}
 
-	err := check.GithubCredentials()
-	if err != nil {
-		return errors.WithMessage(err, "could not check github credential path")
+	e := repo.Entry{
+		Name:      vars.RepoName,
+		LocalPath: vars.RepoPath,
+		URL:       vars.RepoURL,
 	}
 
-	cred, err := github.GetGithubCredentials()
+	// Initializing empty repo with given variables
+	r, err := repo.NewFrameworkRepository(e)
 	if err != nil {
-		return errors.WithMessage(err, "could not get github credential")
+		return errors.WithMessage(err, "could not build framework repository")
+	}
+
+	// Downloading index.yaml file if not existing in current repo path
+	if !r.IndexFile.Exists() {
+		err := r.DownloadIndexFile(vars.RepoPath)
```

Comment:
We could do `file, err := r.DownloadIndexFile(xxx)` and all this logic around will be hidden inside that method. I can propose that change if you want...

Reply:
agreed. push the implementation details of this through 77 up into the repo methods.

Reply:
cool, go for it! 👍